python/vertx/eventbus.py

```diff
@@ -0,0 +1,346 @@
+    """
+    State of EventBus Client
+    """
+    NEW = 0  # when created or not connected / failed
+    CONNECTING = 1  # when the client is connecting to the bridge
+    CONNECTED = 2  # when the client gets connected to the bridge
+    CLOSING = 3  # when the client is closing the connection
+    CLOSED = 4  # when the client closed the connection
+    BROKEN = 5  # when the client connection is broken
+
+
+class EventBus:
+    """
+    Vert.x TCP EventBus Client for Python
+    """
+    
+    def __init__(self, host='localhost', port=7000, options=None, err_handler=None):
+        """
+        EventBus Constructor
+
+        Args:
+            host(str): the host to connect to - default: 'localhost'
+            port(int): the port to use - default: 7000
+            options(dict): e.g. { ping_interval=5, timeout=60, debug=False, connect=False}
+
+        :raise:
+           :IOError: - the socket could not be opened
+           :Exception: - some other issue e.g. with starting the listening thread
+
+        """
+        self.sock = None
+        self.last_pong = None
+        self._state = _State.NEW
+        self.host = host
+        self.port = port
+        self.handlers = {}
+        self.options = options
+        self.timeout = 60  # socket timeout, in seconds
+        self.ping_interval = 5  # heart beat for ping/pong
+        self._err_handler = err_handler
+        self.auto_connect = True
+        self.max_reconnect = 5
+        if self._err_handler is None:
+            self._err_handler = EventBus._default_err_handler
+        if options is not None:
+            if "timeout" in options:
+                self.timeout = int(options["timeout"])
+            if "ping_interval" in options:
+                self.ping_interval = int(options["ping_interval"])
+            if "auto_connect" in options:
+                self.auto_connect = bool(options["auto_connect"])
+            if "max_reconnect" in options:
+                self.max_reconnect = int(options["max_reconnect"])
+            if "connect" in options and bool(options["connect"]):
+                self.connect()
+    
+    @staticmethod
+    def _default_err_handler(message):
+        LOGGER.error(f'Got Error Message: {message} from server')
+    
+    def connect(self):
+        if self._state == _State.CLOSED:
+            LOGGER.debug("Client has been closed")
+            return None
+        num_of_tries = self.max_reconnect if self.auto_connect else 1
+        for i in range(num_of_tries):
+            try:
+                if self._state != _State.CONNECTED:
+                    self._state = _State.CONNECTING
+                    self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                    self.sock.settimeout(self.timeout)
+                    self.sock.connect((self.host, self.port))
+                    self._state = _State.CONNECTED
+                    # receiving thread
+                    t1 = Thread(target=self._receive)
```

Comment:
The variable name is too short. With a better name, the comment becomes unnecessary.